restapi/api/routes.go

```diff
@@ -13,6 +13,9 @@ func registerRoutes(router *gin.RouterGroup) {
 	device.GET("/screenshot", Screenshot)
 	device.PUT("/setlocation", SetLocation)
 	device.POST("/resetlocation", ResetLocation)
+	device.GET("/conditions", GetSupportedConditions)
```

Comment:
I think if you put your three endpoints into an extra routergroup r
and then `r.Use(api.LimitNumClientsUDID())` you should not need a Mutex.